### PR TITLE
Allow vote edit feature

### PR DESCRIPTION
**Pull request by akhoury**
https://github.com/NodeBB/nodebb-plugin-poll/issues/66

screen shot 2019-03-02 at 5 02 07 pm

screen shot 2019-03-02 at 5 25 30 pm

screen shot 2019-03-02 at 5 25 37 pm

screen shot 2019-03-02 at 5 25 42 pm

screen shot 2019-03-06 at 10 19 16 pm

### CODE REVIEW

File: languages/en_GB/poll.json [hunk 1]
```diff
@@ -32,10 +33,13 @@
   "warning.redactor": "You're using Redactor. Do <strong>not</strong> edit the markup manually. Instead, re-open the poll creator.",
 
   "vote": "Vote",
+  "update_vote": "Update vote",
+  "remove_vote": "Remove vote",
   "to_voting": "To Voting",
   "to_results": "Results",
   "voting_unavailable_title": "Voting unavailable",
   "voting_unavailable_message": "This poll has ended or has been marked as deleted. You can still view the results.",
+  "voting_update_disallowed_message": "You already voted and voting update has been disallowed for this poll",
```
Comment (julianlam) [inline languages/en_GB/poll.json]: Suggest: "You have already voted and changing your vote is not allowed for this poll"

File: plugin.json [hunk 2]
```diff
@@ -45,5 +45,9 @@
     "public/js/admin.js"
   ],
   "templates": "templates",
-  "languages": "languages"
+  "languages": "languages",
+  "upgrades": [
+    "upgrades/give_categories_poll_create_privs_to_reg_users.js",
+    "upgrades/convert_poll_id_voters_and_options_votes_to_sorted_sets.js"
+  ]
```
Comment (julianlam) [inline plugin.json]: :+1:

File: templates/poll/view.tpl [hunk 3]
```diff
@@ -10,18 +10,19 @@
         </div>
         <div class="panel-body">
             <div class="poll-view-messages hidden"></div>
```
Comment (julianlam) [inline templates/poll/view.tpl]: How many messages can go in here? I'd suggest maybe showing a warning during voting step to notify users that their vote is final.

Perhaps you can put it in `poll/view/voting`

Comment (akhoury) [reply templates/poll/view.tpl]: Done see screenshot

screen shot 2019-03-06 at 10 19 16 pm

**Comment by akhoury** [comment]
@julianlam are we set to merge here?

**Comment by julianlam** [comment]
:+1: Can you replace string with a different lib? We use validator in NodeBB. Looks like there's a security issue with stringjs

**Pull request merged**